private/buf/bufcli/errors.go

```diff
@@ -164,21 +164,30 @@ func NewTemplateNotFoundError(owner string, name string) error {
 // Note that this function will wrap the error so that the underlying error
 // can be recovered via 'errors.Is'.
 func wrapError(err error) error {
-	if err == nil || (err.Error() == "" && !rpc.IsError(err)) {
-		// If the error is nil or empty and not an rpc error, we return it as-is.
+	if err == nil || (err.Error() == "" && !isConnectError(err)) {
```

Comment:
Something to note about Connect error behavior is `.Error()` prints the code followed by the message.

https://github.com/bufbuild/connect-go/blob/main/error.go#L72-L78

Throughout this PR we want to make sure we're not composing errors with a stutter. I don't think that'll be the case because iirc the errors are hand-crafted.

Also, if you need to get at the underlying error can use `.Message()`:

https://github.com/bufbuild/connect-go/blob/main/error.go#L84-L89

Reply:
In testing, it turns out there were cases where the displayed error would have stuttered such as `Failure: FailedPrecondition: track main has no commits`.

I modified the `wrapError` function here that hopefully fixes this issue.  Now it attempts to convert the given error to a `connect.Error` and then reads the code and message from that.

I think the `if` statement should still catch the same conditions that were previously being caught.  If `err == nil` then the `asConnectError` function will return `nil, false`.  The empty condition should be getting caught via the `asConnectError` function also.

Let me know if I'm missing something though.